mitodl/ocw-hugo-themes
# update home_course_cards with recent changes to level

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/256

#### What's this PR do?
In https://github.com/mitodl/ocw-to-hugo/pull/398 and subsequently https://github.com/mitodl/ocw-to-hugo/pull/399 and https://github.com/mitodl/ocw-studio/pull/757, we updated the `level` property in `ocw-to-hugo`'s output to match changes to `ocw-studio` that allow multiple choices for level and also remove the search URL.  The purpose of `course_legacy.json` is to be able to make changes to the course data template (`course.json`) and not affect the import process in `ocw-studio` unless it's necessary.  Since the structure of `level` has changed, an update was necessary and now that courses processed in this manner are being imported in `ocw-studio`, the API is delivering data that is not currently compatbile with `home_course_cards.html`.  This PR updates the layout to look up the search URL based on the level using the `get_search_url.html` partial and moves some necessary data into `base-theme` so it can be used by both the `www` and `course` themes.

#### How should this be manually tested?
 - Clone this repo on the PR branch
 - In your `.env` file, set `OCW_STUDIO_BASE_URL=http://ocw-studio-rc.odl.mit.edu/`
 - Run `npm run start:www`
 - Verify that the site spins up properly and is available at http://localhost:3000
